### PR TITLE
Tags to template

### PR DESCRIPTION
**Pull request by kuba-mazurkiewicz**
added 2 new resources:

- tag
- assign_templates_to_tag

Assign_templates_to_tag required manual modification in DELETE function in resource file to iterate through templates and destroy them one by one using URI: 
dna/intent/api/v1/tag/{TAGID}/member/{TEMPLATE_ID}

and manual changes in fromBody and updateFromBody functions in model file to support Read operations

### CODE REVIEW

File: gen/definitions/tag.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    response_data_path: response.0.dynamicRules
+    type: List
+    description: Dynamic rules details
+    exclude_test: true
+    attributes:
+      - model_name: memberType
+        type: String
+        description: memberType of the tag (e.g. networkdevice, interface)
+      - model_name: values
+        tf_name: rule_values
```
Comment (danischm) [inline gen/definitions/tag.yaml:37]: I think we can remove the `tf_name` here, as the parent element is already named `dynamic_rules`.

File: gen/definitions/tag.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        data_path: rules
+        type: List
+        element_type: String
+        description: 'values of the parameter,Only one of the value or values can be used for the given parameter. (for managementIpAddress e.g. [\"10.197.124.90\",\"10.197.124.91\"])'
+      - model_name: items
+        tf_name: rule_items
```
Comment (danischm) [inline gen/definitions/tag.yaml:43]: `tf_name` can be removed

File: gen/definitions/tag.yaml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    description: true for system created tags, false for user defined tag
+    example: false
+  - model_name: dynamicRules
+    response_data_path: response.0.dynamicRules
+    type: List
+    description: Dynamic rules details
+    exclude_test: true
+    attributes:
+      - model_name: memberType
+        type: String
```
Comment (danischm) [inline gen/definitions/tag.yaml:34]: should this be an `enum`?

File: gen/definitions/tag.yaml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        element_type: String
+        description: 'values of the parameter,Only one of the value or values can be used for the given parameter. (for managementIpAddress e.g. [\"10.197.124.90\",\"10.197.124.91\"])'
+      - model_name: items
+        tf_name: rule_items
+        data_path: rules
+        type: List
+        description: 'items details, multiple rules can be defined by items'
+        attributes:
+          - model_name: operation
+            type: String
```
Comment (danischm) [inline gen/definitions/tag.yaml:49]: should this be an `enum`?

File: gen/definitions/tag.yaml [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+          - model_name: name
+            type: String
+            description: Name of the parameter (e.g. managementIpAddress,hostname)
+          - model_name: value
+            type: String
+            description: Value of the parameter (e.g. %10%,%NA%)
+      - model_name: operation
+        tf_name: rule_operation
+        data_path: rules
+        type: String
```
Comment (danischm) [inline gen/definitions/tag.yaml:60]: should this be an `enum`?

File: gen/definitions/tag.yaml [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+            type: String
+            description: Operation of the rule (e.g. OR,IN,EQ,LIKE,ILIKE,AND)
+          - model_name: name
+            type: String
+            description: Name of the parameter (e.g. managementIpAddress,hostname)
+          - model_name: value
+            type: String
+            description: Value of the parameter (e.g. %10%,%NA%)
+      - model_name: operation
+        tf_name: rule_operation
```
Comment (danischm) [inline gen/definitions/tag.yaml:58]: `tf_name` can be removed

File: gen/definitions/tag.yaml [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+        data_path: rules
+        type: String
+        description: Operation of the rule (e.g. OR,IN,EQ,LIKE,ILIKE,AND)
+      - model_name: name
+        tf_name: rule_name
```
Comment (danischm) [inline gen/definitions/tag.yaml:63]: `tf_name` can be removed

File: gen/definitions/tag.yaml [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+        data_path: rules
+        type: String
+        description: 'Name of the parameter (e.g. for interface:portName,adminStatus,speed,status,description. for networkdevice:family,series,hostname,managementIpAddress,groupNameHierarchy,softwareVersion)'
+      - model_name: value
+        tf_name: rule_value
```
Comment (danischm) [inline gen/definitions/tag.yaml:68]: `tf_name` can be removed

**Review by danischm — CHANGES_REQUESTED** [review]

File: gen/definitions/tag.yaml (hunk 3 above)
Comment (kuba-mazurkiewicz) [reply gen/definitions/tag.yaml:34]: probably yes, but from documentation its hard to tell what should be valid options, they only gave examples like: networkdevice, interface. But I know there might me some more options to choose from, so I left it as String

File: gen/definitions/tag.yaml (hunk 5 above)
Comment (kuba-mazurkiewicz) [reply gen/definitions/tag.yaml:60]: done

File: gen/definitions/tag.yaml (hunk 1 above)
Comment (kuba-mazurkiewicz) [reply gen/definitions/tag.yaml:37]: done

File: gen/definitions/tag.yaml (hunk 6 above)
Comment (kuba-mazurkiewicz) [reply gen/definitions/tag.yaml:58]: done

File: gen/definitions/tag.yaml (hunk 7 above)
Comment (kuba-mazurkiewicz) [reply gen/definitions/tag.yaml:63]: done

File: gen/definitions/tag.yaml (hunk 8 above)
Comment (kuba-mazurkiewicz) [reply gen/definitions/tag.yaml:68]: done

File: gen/definitions/tag.yaml (hunk 2 above)
Comment (kuba-mazurkiewicz) [reply gen/definitions/tag.yaml:43]: done

**Review by danischm — APPROVED** [review]

**Pull request merged**